### ISSUE TITLE
[EXPERIMENT] Reduce signal:zero weight ratio to combat overfitting

### ISSUE THREAD

**Issue by wildwasser** [wildwasser/sneaker#36]
## Problem

Massive overfitting observed:
- Train Signal R²: 95%
- Test Signal R²: 21%
- Gap: 74 percentage points (catastrophic)
- Direction Accuracy: 3-4% (broken)

## Hypothesis

Current 5:1 signal:zero ratio causes model to memorize rare signal patterns instead of learning generalizable features.

## Proposed Experiments

Test signal-zero ratios:
- Baseline: 5.0 (current)
- Moderate: 4.0
- Significant: 3.0
- Aggressive: 2.0
- Extreme: 1.0 (equal weighting)

## Expected Outcomes

- Lower train R² (60-80% vs 95%)
- Higher test R² (30-40% vs 21%)
- Better generalization
- Reduced overfitting gap
- Improved direction accuracy

## Implementation

Use existing --signal-zero-ratio parameter in scripts/08_train_model.py with Bayesian-optimized hyperparameters from Trial 57.

## Success Criteria

- Test R² improves by 20%+ (0.21 → 0.25+)
- Train/test gap reduces to <30 points
- Direction accuracy improves to >10%

**Comment by wildwasser** [comment]
## ARBUSDT Validation Results (Out-of-Distribution Test)

To determine if poor generalization was pair-specific or fundamental, tested ratio 4.0 model on **ARBUSDT** - a cryptocurrency NOT in the 20 training pairs.

### Validation Pipeline

1. ✅ Downloaded ARBUSDT data (23,265 candles, 2023-2025)
2. ✅ Added macro indicators (BTC_PREMIUM, ETH_PREMIUM, BNB, GOLD)
3. ✅ Feature engineering (89 shared features)
4. ✅ Ghost signal detection (1,234 signals detected)
5. ✅ Created 12-candle windows (23,254 windows total)
6. ✅ Evaluated with ratio 4.0 model

### Results: COMPLETE GENERALIZATION FAILURE

```
Performance on ARBUSDT (unseen crypto):
  Signal R²:          -48.12%  ❌ NEGATIVE (worse than predicting mean\!)
  Direction Accuracy:  49.59%  ❌ Worse than random coin flip (50%)
  Signal MAE:          2.43σ

Comparison to Test Set (20 training pairs):
  Test Set R²:        +18.87%
  ARBUSDT R²:         -48.12%
  Generalization Gap:  -66.99 percentage points  ❌❌❌

Verdict: Model has SERIOUS overfitting to training pairs
```

### Critical Findings

**The problem is NOT signal:zero ratio** - it's fundamental overfitting:

1. **Negative R² (-48%)**: Model predictions are worse than just predicting zero for everything
2. **Random direction (49.6%)**: Model cannot even predict up vs down on new crypto
3. **Massive gap (-67pp)**: Model memorized training pairs instead of learning transferable patterns

### Root Cause Analysis

Signal weight reduction (5→4→3→2) was testing the **WRONG hypothesis**:

- ❌ **What we tested**: Does reducing signal weight reduce overfitting?
- ✅ **What we found**: Model cannot generalize to ANY new cryptocurrency

**Likely root causes:**
- **Too many features** (1,068 features from 89 × 12 windows)
- **Too few training pairs** (only 20 cryptos)
- **Pair-specific memorization** instead of universal pattern learning
- **Model complexity** (511 leaves, depth 13)

### Files Generated

All validation artifacts saved:
- `logs/arbusdt_validation_results.json` - Metrics summary
- `logs/validation_arbusdt_pipeline.log` - Full pipeline log
- `logs/arbusdt_evaluation.log` - Evaluation results
- `data/features/validation_arb_*.json` - Processed ARBUSDT data (966MB windowed)

### Conclusion: Issue #36 Experiments FAILED

**Signal:Zero Ratio Reduction Results:**
| Ratio | Train R² | Test R² | Overfitting Gap | Generalization |
|-------|----------|---------|-----------------|----------------|
| 5.0   | 94.08%   | 20.84%  | 73.24pp         | Baseline       |
| 4.0   | 94.08%   | 18.87%  | 75.21pp (+2pp)  | -1.97pp ❌     |
| 3.0   | 91.79%   | 18.13%  | 73.66pp (+0.4pp)| -2.71pp ❌     |
| 2.0   | 87.74%   | 16.90%  | 70.84pp (-2.4pp)| -3.94pp ❌     |

**ARBUSDT (completely new crypto):**
- Ratio 4.0: **-48.12% R²** (CATASTROPHIC FAILURE)

### Recommendation

**DO NOT pursue signal weight reduction further.** The real problems are:

1. **Need more diverse training data** (expand from 20 to 50+ pairs)
2. **Need feature selection** (reduce from 1,068 features)
3. **Need simpler models** (reduce leaves/depth)
4. **Need better validation** (always test on out-of-distribution cryptos)

Closing this issue. Next focus should be:
- Issue #38: Feature selection (reduce dimensionality)
- Issue #37: Aggressive regularization (simpler models)
- **NEW**: Expand training pairs from 20 to 50+ cryptocurrencies
